### PR TITLE
Aoaf 263 be create java 21 app including ecs terraform for the presenter account api

### PR DESCRIPTION
**Pull request by myadavch**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+				<artifactId>dependency-check-maven</artifactId>
+				<version>${dependency-check-plugin.version}</version>
+				<executions>
+					<execution>
+						<goals>
+							<goal>check</goal>
+						</goals>
+					</execution>
+				</executions>
+			</plugin>
```
Comment (amartin7663) [inline pom.xml:266]: The Owasp plugin is in the parent pom now, so if you upgrade to 2.1.4 you don't need this anymore. Example: https://github.com/companieshouse/accounts-association-api/pull/3

File: src/test/java/uk/gov/companieshouse/presenter/account/PresenterAccountApiApplicationTests.java [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package uk.gov.companieshouse.presenter.account;
+
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
+class PresenterAccountApiApplicationTests {
```
Comment (amartin7663) [inline src/test/java/uk/gov/companieshouse/presenter/account/PresenterAccountApiApplicationTests.java:6]: You don't need the this to plural right?
```suggestion
class PresenterAccountApiApplicationTest {
```

File: src/test/java/uk/gov/companieshouse/presenter/account/controller/PresenterAccountControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.HttpStatus;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+
+
+@ExtendWith(MockitoExtension.class)
```
Comment (amartin7663) [inline src/test/java/uk/gov/companieshouse/presenter/account/controller/PresenterAccountControllerTest.java:15]: Do you need this if you're not actually mocking anything?

File: src/test/java/uk/gov/companieshouse/presenter/account/controller/PresenterAccountControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+public class PresenterAccountControllerTest {
+
+    @InjectMocks
```
Comment (amartin7663) [inline src/test/java/uk/gov/companieshouse/presenter/account/controller/PresenterAccountControllerTest.java:18]: Same with this?

File: terraform/groups/ecs-service/locals.tf [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    trimprefix(sec.name, "/${local.name_prefix}/") => sec.arn
+  }
+
+  service_secrets_arn_map = {
+    for sec in module.secrets.secrets :
+    trimprefix(sec.name, "/${local.service_name}-${var.environment}/") => sec.arn
+  }
+
+  task_secrets = [
+    { "name" : "CHS_API_KEY", "valueFrom" : "${local.service_secrets_arn_map.chs_api_key}" },
```
Comment (amartin7663) [inline terraform/groups/ecs-service/locals.tf:56]: Can we rename this variable PRESENTER_ACCOUNT_API_KEY, we should be separate api keys for each api. We should have done this in the accounts-filing-api also

File: terraform/groups/ecs-service/main.tf [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+  docker_registry   = var.docker_registry
+  docker_repo       = local.docker_repo
+  container_version = var.presenter_account_api_version
+  container_port    = local.container_port
+
+  # Service configuration
+  service_name = local.service_name
+  name_prefix  = local.name_prefix
+
+  # Service Healthcheck configuration
```
Comment (amartin7663) [inline terraform/groups/ecs-service/main.tf:61]: Shouldn't this be on line 46?

File: terraform/groups/ecs-service/locals.tf [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+  eric_secrets = [
+    { "name" : "AES256_KEY" , "valueFrom" : "${local.service_secrets_arn_map.aes256_key}" },
+    { "name" : "API_KEY" , "valueFrom" : "${local.service_secrets_arn_map.api_key}" },
+    { "name" : "CACHE_URL" , "valueFrom" : "${local.service_secrets_arn_map.cache_url}" }
+  ]
+
+  eric_environment = [
+    { "name": "LOGLEVEL", "value": "${var.log_level}" },
+    { "name": "MODE", "value": "api" },
+    { "name": "PRESENTER_ACCOUNT_URL", "value" : "${var.presenter_account_url}" },
```
Comment (amartin7663) [inline terraform/groups/ecs-service/locals.tf:76]: This should be the account_api_url, it's used by ERIC to check api keys and oauth2 log ins.

File: terraform/groups/ecs-service/locals.tf [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+  }
+
+  service_secrets_arn_map = {
+    for sec in module.secrets.secrets :
+    trimprefix(sec.name, "/${local.service_name}-${var.environment}/") => sec.arn
+  }
+
+  task_secrets = [
+    { "name" : "CHS_API_KEY", "valueFrom" : "${local.service_secrets_arn_map.chs_api_key}" },
+    { "name" : "INTERNAL_API_URL", "valueFrom" : "${local.service_secrets_arn_map.internal_api_url}" },
```
Comment (amartin7663) [inline terraform/groups/ecs-service/locals.tf:57]: I don't think we'll ever need to call an api from this service so we don't need this config

File: terraform/groups/ecs-service/locals.tf [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+    { "name" : "MONGODB_URL", "valueFrom" : "${local.service_secrets_arn_map.mongodb_url}" }
+  ]
+
+  task_environment = [
+    { "name" : "API_URL", "value" : "${var.api_url}" },
```
Comment (amartin7663) [inline terraform/groups/ecs-service/locals.tf:62]: Same here, we won't need this config

File: terraform/groups/ecs-service/variables.tf [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+}
+variable "required_memory" {
+  type        = number
+  description = "The required memory for this service"
+  default     = 256 # defaulted low for node service in dev environments, override for production
+}
+variable "use_fargate" {
+  type        = bool
+  description = "If true, sets the required capabilities for all containers in the task definition to use FARGATE, false uses EC2"
+  default     = false
```
Comment (amartin7663) [inline terraform/groups/ecs-service/variables.tf:47]: Can we try setting this to true, we are supposed to be using fargate.

File: pom.xml (hunk 1 above)
Comment (myadavch) [reply pom.xml:266]: update the parent file as suggested.

File: src/test/java/uk/gov/companieshouse/presenter/account/PresenterAccountApiApplicationTests.java (hunk 2 above)
Comment (myadavch) [reply src/test/java/uk/gov/companieshouse/presenter/account/PresenterAccountApiApplicationTests.java:6]: done.

File: src/test/java/uk/gov/companieshouse/presenter/account/controller/PresenterAccountControllerTest.java (hunk 4 above)
Comment (myadavch) [reply src/test/java/uk/gov/companieshouse/presenter/account/controller/PresenterAccountControllerTest.java:18]: yes it is not required.

File: terraform/groups/ecs-service/locals.tf (hunk 5 above)
Comment (myadavch) [reply terraform/groups/ecs-service/locals.tf:56]: updated .

File: terraform/groups/ecs-service/main.tf (hunk 6 above)
Comment (myadavch) [reply terraform/groups/ecs-service/main.tf:61]: yes, thanks for spotting.

File: terraform/groups/ecs-service/locals.tf (hunk 7 above)
Comment (myadavch) [reply terraform/groups/ecs-service/locals.tf:76]: renamed to `account_api_url`

File: terraform/groups/ecs-service/locals.tf (hunk 8 above)
Comment (myadavch) [reply terraform/groups/ecs-service/locals.tf:57]: removed

File: terraform/groups/ecs-service/locals.tf (hunk 9 above)
Comment (myadavch) [reply terraform/groups/ecs-service/locals.tf:62]: Remove.

File: terraform/groups/ecs-service/variables.tf (hunk 10 above)
Comment (myadavch) [reply terraform/groups/ecs-service/variables.tf:47]: It is set to true.

File: src/test/java/uk/gov/companieshouse/presenter/account/controller/PresenterAccountControllerTest.java (hunk 3 above)
Comment (myadavch) [reply src/test/java/uk/gov/companieshouse/presenter/account/controller/PresenterAccountControllerTest.java]: yes it is not required.

**Review by amartin7663 — APPROVED** [review]

**Pull request merged**